cfme/intelligence/chargeback.py

```diff
@@ -144,119 +178,69 @@ def _fill_assignform_dict(form, d):
-                 compute_fixed_2=None,
-                 mem_alloc=None,
-                 mem_used=None,
-                 mem_used_var=None,
-                 net_io=None,
-                 net_io_var=None,
-                 appliance=None):
+                 currency=None,
+                 fields=None,
+                 appliance=None,
+                 ):
         Navigatable.__init__(self, appliance=appliance)
         self.description = description
-        self.cpu_alloc = cpu_alloc
-        self.cpu_used = cpu_used
-        self.cpu_used_var = cpu_used_var
-        self.disk_io = disk_io
-        self.disk_io_var = disk_io_var
-        self.compute_fixed_1 = compute_fixed_1
-        self.compute_fixed_2 = compute_fixed_2
-        self.mem_alloc = mem_alloc
-        self.mem_used = mem_used
-        self.mem_used_var = mem_used_var
-        self.net_io = net_io
-        self.net_io_var = net_io_var
+        self.currency = currency
+        self.fields = fields
 
     def create(self):
-        navigate_to(self, 'New')
-        fill(rate_form,
-            {'description': self.description,
-             'cpu_alloc': self.cpu_alloc,
-             'cpu_used': self.cpu_used,
-             'cpu_used_var': self.cpu_used_var,
-             'disk_io': self.disk_io,
-             'disk_io_var': self.disk_io_var,
-             'compute_fixed_1': self.compute_fixed_1,
-             'compute_fixed_2': self.compute_fixed_2,
-             'mem_alloc': self.mem_alloc,
-             'mem_used': self.mem_used,
-             'mem_used_var': self.mem_used_var,
-             'net_io': self.net_io,
-             'net_io_var': self.net_io_var},
-            action=rate_form.add_button)
-        flash.assert_no_errors()
+        view = navigate_to(self, 'New')
+        fields = self.fields
+        view.fill_with({'description': self.description,
+                        'currency': self.currency,
+                        'fields': fields},
+                       on_change=view.add_button,
+                       no_change=view.cancel_button)
+
+        view.flash.assert_success_message('Chargeback Rate "{}" was added'.format(
+            self.description))
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        fill(rate_form,
-            {'description': updates.get('description'),
-             'cpu_alloc': updates.get('cpu_alloc'),
-             'cpu_used': updates.get('cpu_used'),
-             'disk_io': updates.get('disk_io'),
-             'compute_fixed_1': updates.get('compute_fixed_1'),
-             'compute_fixed_2': updates.get('compute_fixed_2'),
-             'mem_alloc': updates.get('memory_allocated'),
-             'mem_used': updates.get('memory_used'),
-             'net_io': updates.get('network_io')},
-            action=rate_form.save_button)
-        flash.assert_no_errors()
+        view = navigate_to(self, 'Edit')
+        view.fill_with(updates, on_change=view.save_button, no_change=view.cancel_button)
```

Comment:
as far as I see you have moved most class properties to fields dict and pass it as a param. it should have mappings between view and values and it should be defined everywhere in tests.
I wouldn't say I like it. Lets listen to other opinions.